File: ProtoScaner.Server/Models/HistorialPacienteIngreso.cs
using System;
using System.Collections.Generic;

namespace ProtoScaner.Server.Models;

public partial class HistorialPacienteIngreso
{
    public int IdHistorial { get; set; }

    public int? IdPaciente { get; set; }

    public int? TipoAmputacion { get; set; }

    public int? LadoAmputacion { get; set; }

    public DateOnly? FechaAmputacion { get; set; }

    public int? Causa { get; set; }

    public bool? Terapia { get; set; }

    public string? TiempoTerapia { get; set; }

    public int? IdMedida { get; set; }

    public string? Comentario { get; set; }

    public virtual CausaAmputacion? CausaNavigation { get; set; }

    public virtual Paciente? IdPacienteNavigation { get; set; }

    public virtual LadoAmputacion? LadoAmputacionNavigation { get; set; }

    public virtual TipoAmputacion? TipoAmputacionNavigation { get; set; }
}

File: protoscaner.client/src/types/HistorialPacienteIngreso.ts
// src/types/HistorialPacienteIngreso.ts
export interface HistorialPacienteIngreso {
    idHistorial: number;
    idPaciente?: number;
    tipoAmputacion?: number;
    ladoAmputacion?: number;
    fechaAmputacion?: string; // Date en TypeScript, pero con el tipo string para JSON compatible
    causa?: number;
    terapia?: boolean;
    tiempoTerapia?: string;
    idMedida?: number;
    comentario?: string;
}


// src/types/HistorialPacienteIngreso.ts

export interface HistorialPacienteIngresoDTO {
    IdHistorial: number;
    IdPaciente: number;
    TipoAmputacion: number; // 1: Transtibial, 2: Transfemoral
    LadoAmputacion: string;
    FechaAmputacion: string; // ISO Date string
    Causa: string;
    Terapia: string;
    TiempoTerapia: number;
    IdMedida: number;
    Comentario: string;
}
